gossip/src/cluster_info.rs

```diff
@@ -1045,6 +1045,31 @@ impl ClusterInfo {
         }
     }
 
+    fn find_oldest_vote_index(&self, should_evict_vote: impl Fn(&Vote) -> bool) -> u8 {
+        let self_pubkey = self.id();
+        let mut num_crds_votes = 0;
+        let vote_index = {
+            let gossip_crds =
+                self.time_gossip_read_lock("gossip_read_push_vote", &self.stats.push_vote_read);
+            (0..MAX_LOCKOUT_HISTORY as u8)
+                .filter_map(|ix| {
+                    let vote = CrdsValueLabel::Vote(ix, self_pubkey);
+                    let vote: &CrdsData = gossip_crds.get(&vote)?;
```

Comment:
Not sure if there's a good way to address this, but after a restart we could attempt to refresh a vote before our gossip crds is fully populated. in this case we could be evicting a later vote.

I think it's fine anyway, since with `VoteStateUpdate` since *most* of the time only the last vote matters, but just wanted to point out that we might not always evict the "oldest" vote here.

Reply:
yeah, it will be some vote "older" than the newest vote because to evict, you have to have >=32 votes in your gossip table

Reply:
yea I think that's true as long as the tower file isn't corrupt which we shouldn't worry about.